### DISCUSSION TITLE
Fullbody IK

### DISCUSSION THREAD

**Discussion by AMZN-Liv** [o3de/rfcs#20] [Ideas]
**Is your feature request related to a problem? Please describe**
Within the current EMFX system, there is support for two types of IK, Aim IK and Look IK. The current solution for inverse kinematics is limited in the scope of movement that can be derived, and it is hard to author character movement that naturally applies movement beyond linked joint rotations. Fullbody IK is also useful for fully tracked avatar movements that mirror a human action, such as someone with a motion capture suit or using VR trackers.

**Describe the solution you'd like**
A fullbody IK solution would make it possible to author more realistic character movements that utilize the entire body of the character rig. An example of the type of movement that fullbody IK enables is bending over or squatting down to interact with an object.

**Comment by GaianHelmers** [comment]
I came upon this RFC from google and definitely agree.

I've been doing some looking into Open"er" solutions and came upon this IK library: https://github.com/TheComet/ik
I'm not sure of the legalities of making a "Comet IK Gem" with this, nor the logistics of making it O3DE compatible. But I wanted to bring this up as a lead for finding a resolution to this.
Natural feeling IK is a critical part of character performance, I'd love to have that box checked in my laundry list of needs.
